### DISCUSSION TITLE
Target rate sensor - maximise earnings instead of minimising import rates

### DISCUSSION THREAD

**Discussion by ccristal** [BottlecapDave/HomeAssistant-OctopusEnergy#430] [Ideas]
Hi!
Apologies, I now realise my previous post #424 didn't make much sense... :-)
Hopefully this one explains better what I have in mind.

This applies to intermittent slot selection.
Think about users who have Agile for both import and export. 
Basically, for every 1/2 hour slot, I need to decide whether to use it or export it. I want to use if it's "the cheapest", otherwise I would be better off exporting it. 

The problem is the meaning of "cheapest". At the moment, your automation selects the _cheapest_ slots based on import rate only. My point is that it should, instead, choose them based on the difference between import rate of the slected slots and export rate of all the other slots in the selection period, and select the slots that minimise that difference.

So, suppose I have two 1/2 hour slots A and B, and that I need 1/2 hour to run my appliance:

|Slot|Import rate|Export rate|
| --- | --- | --- |
|A|1p|10p|
|B|2p|5p|

Your current sensor would activate on slot A. But it would make much more sense exporting slot A and running my appliance on slot B: yes, I would pay more, but I would be better off overall, earning net 8p over the whole period instead of 4p.

Does it make sense?

Thanks! :-)

**Comment by BottlecapDave** [comment]
Afternoon, and yeah that makes more sense and a useful addition :) 

I'll need to think about how this would fit into the target rate sensors as a whole as they're getting quite chunky with all of the configuration. Bugs and other long running features are currently my highest priority, so no promise when I'd get to this.
